### ISSUE TITLE
All VCC Sites - Billing Codes Updates

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1806]
In GitLab by @kdjstudios on Nov 4, 2019, 13:00

**HD:** http://www-servicedesk.answernet.com/profiles/ticket/2019-11-05-23323/conversation

**HD:** http://www-servicedesk.answernet.com/profiles/ticket/2019-11-26-15455/conversation - Closed

Hello Team,

During our discussion with Rich and Gary about the revenue vs non-revenue concern we have an updated plan below. Please let us know if there are any questions or concerns.  

**Goal 1:** - Completed

1. **[Completed]** Reverse all updates to revenue/non-revenue codes. This is to the reports and the site edit page. The original implementation of this was not done per Gary standards. Please have these updates reversed ASAP.

**Goal 2:** - In Progress

1. **[Completed]** List of currently used codes in the last 12 months. Grouped by Site. (This way we know what sites are actually using.) 
2. **[Completed]** List of currently 'active' codes, that have not been used in the last 12 months. Grouped by site. (This way we know what codes are not being used and can most likely be deactivated.)
3. **[Completed]** Go code by code  on the lists and figure out what each one is used for and if it needs to be replaced or deactivated.  
4. **[In Progress]** Write a document and training material as to what each code is used for and when to use it. Also distinguish if the code is revenue or non revenue. (Gary sent an email to start addressing this)
5. Complete an internal "Company Codes" documentation to distribute to all of operations.

**Goal 3:** - Waiting on 2

1. **[Completed]** Send an email with notification of the updates to codes and include the documentation from Goal 2.
2. **[Completed]** Provide a list of all sites that have codes specified to be deactivated that are active on accounts. Review with Gary on how to proceed. 
3. **[Completed]** Deactivate codes that are not in use. Replace/update active codes for all sites to use the "Company Codes" list above. 
3. **[Waiting until release]** Confirm with all sites they are able to use the new codes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 4, 2019, 13:01

@amishra @smasih 

Please let us know if you can accomplish Goal 1 overnight, or if it would be best to do during the upcoming release.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 4, 2019, 13:03

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 4, 2019, 13:03

@jredfield 

You mentioned on the call you are able to do the queue to get the lists of codes (Goal 2 steps 1&2). May you please post those here.

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 4, 2019, 13:43

What about distinguishing codes used since VCC conversion--codes needed (or added) since then, and codes no longer needed as a result of the conversion?  I know Gary mentioned that during our meeting today.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 4, 2019, 14:26

@jredfield 

That is what you need to find out by performing steps 1 & 2 of Goal 2. With both of those lists we can then bring that back to operations.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 5, 2019, 06:49

@kdjstudios @jredfield 

Thank you for the update.

Jhon Request via email :
 > Would it be possible to get an updated list of just the Answernet active codes (not the SAHosted ones), that have been in use during the last 12 months?  In other words, Gary wants to remove from that list any codes that have not been used in the last 12 months.

* We ran the scripts for active sites only and attached the file below.

[Activities List OF Answernet Server](/uploads/bfdcdac244899ded5f135fd0bf4cbd0c/new_list_of_all_active_site_activity_last_12_month.csv)

* You can filter with Site Name and Active Since 12 months column, Fee Status and so on.. .


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 5, 2019, 07:10

@smasih @amishra Thank you for the last spreadsheet you posted.  Please clarify the "Active Since 12 months" column in the spreadsheet.  Does "Not Active" mean currently not active, or never used at any time in the last 12 months?

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 5, 2019, 07:23

@smasih Thank you for clarification. "Not Active" does mean never used in last 12 months.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2019, 11:12

I wrote:

> After further review with Gary, it was mentioned that operations should have documentation on what their site’s codes are and what they are being used for. If each of the plus ones could please provide us with their existing documentation it will be extremely helpful when working with Rich and Gary on consolidating all the codes for all sites to use into a standard set of codes moving forward.  NOTE: We are hoping to have this documentation by the end of the week.
> 
>  
> 
> If needed, we do have a list of all codes for each site that we can provide to the plus ones to aid in knowing which codes are currently active in the system and if they have been used in the last 12 months.
> 
>  
> 
> Thank you and please let us know if there are any questions or concerns.

@jredfield 

I think we are all queued up for communicating with Ops on the Goal 2.

@smasih @amishra 

We are still needing to have your team perform Goal 1. Please let us know if this can be completed overnight or if it should be performed during a release.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2019, 12:08

@amishra @smasih @jredfield 

Is that list only the activities? If so we need one for the additional fees too.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2019, 14:04

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2019, 14:05

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 5, 2019, 17:47

@kdjstudios @amishra @smasih Yes it appears that this latest list is only activities.  I don't see additional fees that were present on the previous sheet from http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1590#note_49384. If you could update this latest list or create a new one with additional fees indicating if they were or were not ever used in the last 12 months, that would be great! If you just do a separate list with only additional fees, I can merge it into the previous sheet.

Also in your latest list, there are quite a few codes with numeric prefixes like 053 (e.g. "053 Calls Charge 0440" for Atlanta) that were part of an older billing system before SAB. Why are these still showing as active within the last 12 months? They don't show up on your previous list of currently active codes, so perhaps they were discontinued, but still were used at some point within the last 12 months. So I don't see any reason to use them.  Will need to discuss.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 6, 2019, 03:19

@kdjstudios @jredfield 

Thank you for the update.

> Also in your latest list, there are quite a few codes with numeric prefixes like 053 (e.g. "053 Calls Charge 0440" for Atlanta) that were part of an older billing system before SAB. Why are these still showing as active within the last 12 months? They don't show up on your previous list of currently active codes, so perhaps they were discontinued, but still were used at some point within the last 12 months. So I don't see any reason to use them. Will need to discuss.

* Activity `0447` inactive on the site level. 
* There were 15 accounts and activity `0447` was active for all accounts. [Account List here](/uploads/7f74d8a2ed884438e81e6145a9ddc893/Atlanta_site_activity_0447.ods)
* If any activities are present on the account level then the transactions are created for each activity. 
* As per the current functionality of SAB While creating transactions,  we never check the activity is active or deactivate.
 
* Before we were fetching the code list from the Transaction Records. Now, We are going to get the records(list) from the invoices because they only create transaction line items if an activity is active on both levels (Site and Account.)

> We are still needing to have your team perform Goal 1. Please let us know if this can be completed overnight or if it should be performed during a release.

* We will release this on the upcoming Saturday(**09/11/2019**). If we need to release as an emergency then we can do it tomorrow.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 6, 2019, 03:42

@kdjstudios @jredfield 

In short, what Ashish is trying to say is that we used Usage Transactions table (Usage History) to determine Used Code within 12 months. Which gave us a list but maybe not 100% right as you pointed out (activity 0447). 
Now we are using Invoice to determine used code within 12 months and will post the list for Additional fees and activity (updated activity list) shortly

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 6, 2019, 04:15

@kdjstudios @jredfield 

Please find the below attachment for both Additional Fee & Updated Activity Fee .

[fees_active_from_last_year](/uploads/28c0d3e314776efb653e2a1811982fa4/new_list_of_additional_fees_active_from_last_year.ods)  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 6, 2019, 13:47

@amishra 

Thank you for the updated list and clarifications. We should be able to queue up the removal for this Saturday release. 

@jredfield 

I believe I saw you sent the email with the updated list out as requested by Gary. Is there anything else needed to be done, until we hear back from them?

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 6, 2019, 13:56

@kdjstudios I'll follow up with Gary to see if he wants another meeting to discuss rev/non-rev codes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 6, 2019, 13:59

@jredfield 

I believe we should not be discussing rev/non-rev yet. I believe we still need ops to send us back the "company codes" that all sites should be using, as part of that they will be deciding if a code is rev or non-rev. Right?

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 6, 2019, 14:00

@kdjstudios I did enhance the list somewhat, presenting the data in different ways--multiple worksheet tabs. I sent it to Gary, Rich and a few others. It's here: [SAB_BillingCodes_110619.xlsx](/uploads/151178ddee74ddcdc96396658ef6d8c2/SAB_BillingCodes_110619.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 6, 2019, 14:02

@kdjstudios 

Kyle wrote:

>I believe we should not be discussing rev/non-rev yet. I believe we still need ops to send us back the "company codes" that all sites should be using, as part of that they will be deciding if a code is rev or non-rev. Right?

Yes that was mentioned--you're right.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 8, 2019, 24:39

@kdjstudios @jredfield 

* We are going to release `GOAL 1` tomorrow on the production server.

* In GOAL2, we have already provided the list of active & Inactive code within 12 months(As pointed  1 & 2 )

As per my best understanding, nothing has to do from my end until the rest of the points completed in GOAL 2.

Please Update the labels.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 8, 2019, 06:22

@amishra @kdjstudios @smasih Rashmi, thank you for this update. Yes, we still need to meet with Gary and Rich to determine which codes are to be deactivated and determine which are revenue and non-revenue. The documentation needs to be continued on our side.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 8, 2019, 08:45

@amishra @jredfield 

Yes, please release Goal 1. Then we will mark this ForAN, until the Aavaz team is needed.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 9, 2019, 03:26

@kdjstudios @jredfield 

Thank you for the update.

We have successfully released on both of the production servers. Please have a look.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2019, 10:31

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2019, 11:19

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2019, 11:46

1. Need to schedule meeting with Gary.
 - Find meeting times for next week and confirm participants. 

Latest List From Cori:

[SAB+Activities++Fees+RPM+updated+by+Cori+11-26-19_V3.xlsx](/uploads/576ae7fba6b621d585f96ffaed48813f/SAB+Activities++Fees+RPM+updated+by+Cori+11-26-19_V3.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 9, 2019, 16:22

For Gary:

We have the list of codes from Cori. We now need to know if we are ready to schedule a meeting to review these codes with who: Plus ones, Cori, Michelle, Rich, etc.? also is there anything else we should be addressing prior.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2019, 10:39

@jredfield  have you been able to schedule a meeting with Gary (and the others) to review this?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2020, 09:54

@jredfield 

We will need to have you recap what Gary discussed on the call and post an update here.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2020, 09:57

assigned to @jredfield

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2020, 10:23

@jredfield may you please reach out to Leigh to schedule a meeting (2 hours preferably) with Rich, Cori, Gary, yourself, and me. If we can schedule this within the next two weeks would be amazing. 

Agenda:

1. Go code by code on the "Recurring & Other Fee Code" list provided by Cori to figure out what each one is used for and if it needs to be replaced or deactivated.
1. Document which each code is and how is it used. This will be then used to update the training material.
1. Also if possible we can distinguish if the code is revenue or non revenue. 
1. After the meeting we can then create the complete list of internal "Company Codes" and their documentation to distribute to all of operations.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2020, 10:29

Here is the list of codes sent in from Rich, we will want to merge this as part of the call with the "Company Codes" list. 

[Rich_s_Code_List.xlsx](/uploads/19c28ec5607185deb275ba911dc72a49/Rich_s_Code_List.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 29, 2020, 14:08

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 4, 2020, 12:25

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 4, 2020, 12:26

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 6, 2020, 13:30

This was reviewed with Gary, Rich, and Cori. The following are the updated codes that we will have all sites use and their settings. 

[SAB_Company_Codes__Internal_Sites_.xlsx](/uploads/5eee81e5058930b319ab9d10a4397ce3/SAB_Company_Codes__Internal_Sites_.xlsx)

The above list is still needing to be reviewed with @jredfield and I as at the end of the call Gary stated he had specified which codes were Revenue vs non revenue, but I do not recall hearing anything about that for each code. Unless when he mentioned the code effects AR, as being revenue. 


These are the notes taken during the call also. Each will be opened into a new ticket. 

* General notification to be emailed out when a credit is entered to send an internal notification.
* Would like to add some sort of sign off/approval through out the billing process. 
* All manually entered prompts or Wizards will need to also prompt for additional description to be required. Save the username of user who performed this. 
* Update the SBJ since both the credit and debit are on the invoice it shows as a net "0", but we want this broke down to show as both a credit and debit. This will need to report the positive as revenue, while the negative as credit. 
* Rich would like to setup conversation to review the SCB/S report. Setup a separate meeting to address concerns there. 
* All "Non-recurring" will be same code, but can also be seen in the "additional fees drop down" and staged fees lists.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 18, 2020, 13:15

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 21, 2020, 08:12

did these new tix get opened/worked?

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 21, 2020, 08:43

greg to follow up with gary, rich, and cori

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 27, 2020, 10:30

unassigned @jredfield

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 27, 2020, 11:40

We will want to have Gary confirm which codes were Revenue vs non revenue. We just want to make sure our "keyterms" match.  I believe all "Revenue" codes are the ones that were specified that effected AR, while all other codes would be considered "Non-Revenue". This will be to update the existing codes.

in-regards to the functionality to have ops specify/modify these codes, are we going to use what was previously developed or are we designing this functionality differently?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 2, 2020, 09:40

This was reviewed today and the revenue vs non revenue codes and reporting updates is going to be part of a separate project, where we will be creating "wizards" to handle most action items. This also will link with the Project Deposits (Not Security Deposits). 

This put this back into ensureing that all sites will have the same setup of "Standard Codes" as listed in the above mentioned spreadsheet: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1628#note_55531. 

I think we will need to discuss the best way to implement and test this now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2020, 09:04

This was reviewed today and we will want to start developing anything that is needed to perform this update. As of now we expect the list in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1628#note_55531 is correct, But I will reach out to Cori, Rich, and Gary to confirm.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Jun 24, 2020, 06:34

are the codes/spreadsheet confirmed?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2020, 08:58

@Robleski 

Thank you for the reminder, I had the email still in my drafts. I will be sending it out today.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2020, 11:07

I wrote:

> We are following up from about four months ago when we reviewed the list of Billing Codes for all Internal VCC Sites (excluding NCSM, SA Hosted, Outbound sites, etc.). After this Saturday's  release we believe SA Billing will now be in a position for us to be able to perform these updates for each VCC site.
> 
> We have attached in the full list here for your full review and approval. Please thoroughly review and check this list as this has MAJOR effects on all sites. Please keep in mind that this list is from 4 months ago, so if operations has made updates or changes during that time we may also need to update the list accordingly. NOTE: The Calculations for activities we are not changing. This is simply deactivating fees not on the list and adding fees that are missing on site that are on the list. Also if a fee exists, but has a different billing code or name, we will be updating it to match the list. 
> 
> We currently would like to plan to release these updates on the 11th of July if we receive your approval before the July 7th to allow us time to send out the newsletter and announcement to sites. We thank you in advance and please let us know if there are any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Jun 29, 2020, 07:02

Avaaz waiting on An decesioning

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 29, 2020, 08:05

Cori wrote:

> Hi All- Received. Will review in detail next week. Cori

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2020, 07:41

Cori wrote:

> 1- VCC SCC Transaction Count 55030 was missing from the Activity code list.
> 
> 2- Changed the Name of "TextGen" 120 Recurring/Other code to: "SA ITR"
> 
> 3- Outbound Hours 33040 was missing from the Recurring/Other list.
>
> [All+Internal+VCC+Sites+-+Billing+Codes_CAB__1_.xlsx](/uploads/96ad012ec0a91e6775115043b7153abb/All+Internal+VCC+Sites+-+Billing+Codes_CAB__1_.xlsx)

I wrote:

> We have provided the updated list to the support team. We are checking to see if the support team is able to check to confirm if there are any fees that are not included on this list that operations is currently using. As those will need to be removed from the accounts before the Support team is able to deactivate them at the site level. We will follow up with the support teams response once we receive it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2020, 07:42

@amishra @ranand @Robleski 

Is there a way for us to check which codes that are not on the list above (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1628#note_60626) that are being used by operations on sites using the VCC posting scripts?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 5, 2020, 23:54

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 6, 2020, 24:44

@kdjstudios @Robleski 

Thank you for the update.

* We can check for the VCC activities code only because we have a standard format of it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 6, 2020, 14:33

@amishra @Robleski @rranjan 

That doesn't make sense? This is exactly what we would do when we release this to production, but we just want to test it before on a dev server as to determine what needs to be fixed before we do it on production?

To my knowledge, SAB will notify us and prevent a user from deactivating a fee. so wouldn't you simply need take a image from production and then create a "script" (if that is time saving, vs manually deactivation each code in the UI) to deactivate all those fees not on the list? If we get any emails back we then will have which codes are being used and also which accounts they are being used on, is that correct?

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 7, 2020, 04:50

@kdjstudios  @Robleski 

We have attached the list of Activity/Additional Fee which is present on site but not on account . We have find this for **Active Site** and **Active Account**.

Answernet
[answernet__1628_-_Sheet1.csv](/uploads/a323e88b5b9b99d36a0c614ae326c357/answernet__1628_-_Sheet1.csv)

Keener
[keener__1628_-_Sheet1.csv](/uploads/7dd04e0d449f4098e6523b50870b10b7/keener__1628_-_Sheet1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 7, 2020, 08:52

@rranjan @ranand @amishra  @Robleski @aznbyte 

Either that is the wrong list, or I do not think you understand this request, as those lists contain inactive fees and also appears to be for non VCC sites. In summary, We need to know only  which fees are **active** on only **active** accounts that are **not** on the list for **only** VCC sites.

1. We need the list for only VCC Sites on the internal server. No external server, because there are no VCC sites.
2. The list needs to contain: Account Name, Account Number, Activity/Fee Name, Activity/Fee Billing Code
3. The list will be of all codes that are **NOT** on the list in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1628#note_60626 but are active on an account.
4. The list you can generate by matching the **billing code**, not the name.


So for example, lets say on billing site I find "old fee - 12345" and "older activity - 654987". Both of these billing codes are active on the site, so I need to deactivate them because they are not on the all codes list above. (This is the whole purpose of this ticket, to deactivate all unneeded codes.) 
- So when deactivating "old fee - 12345" I get an error stating it is being used on accounts. That list of accounts is what I need. We need this so we can tell operations to either remove(deactivate) or replace that fee on each of those accounts that it is being used on. So this way when we deactivate it on production we do not encounter this error again.
- So when deactivating "older activity - 654987" I do not get an error, that is good that means that we will be able to deactivate this when releasing to production with out issue. 

Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 9, 2020, 05:15

@kdjstudios  @Robleski 

Thanks for the clarification.

Please find the attached list for the VCC site which is not on the list.


[_1628_answernet_code_not_on_list_-_Sheet1__1_.csv](/uploads/7fd47eb0696f08767cc32ed95a716069/_1628_answernet_code_not_on_list_-_Sheet1__1_.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 10, 2020, 08:47

I wrote:

> The support team was able check the VCC sites for fees that are not part of this list but are being used by operations on accounts. We have attached the list of accounts and their billing codes. In order for the support team to deactivate these fees at the site level operations will need to first remove them from each account. Please advise on how we should be notifying operations to remove/replace these fees. 

@rranjan @Robleski 

I just want to check that the team ran this for all internal sites, and not just the VCC sites as per the request? I noticed both the NCSM and SA Hosted sites were on the list, but those both are not using the VCC switch.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 11, 2020, 05:35

@kdjstudios  @Robleski 


> I just want to check that the team ran this for all internal sites, and not just the VCC sites as per the request? I noticed both the NCSM and SA Hosted sites were on the list, but those both are not using the VCC switch.

 My bad I forgot to filter out the non-vcc site.We have filter non-vcc out and attached.

[code_not_on_list_vcc__1628_-_Sheet1.csv](/uploads/3576a59ee54b28c0ad987b252f09af14/code_not_on_list_vcc__1628_-_Sheet1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:28

I wrote:

> We just wanted to follow up to see how you would like to notify operations of the Billing codes that needs to be updated on the accounts?
> 
> NOTE: The current plan would be to have these completed by this Friday 7/17/2020. This would allow for the support team to release the updates over the weekend.

Cori wrote:

> I'll work with the Sites on this so you guys can just hold tight. I want to be sure there aren't additional codes we need to create to accommodate these codes- or just replace with a code we already have. We can shoot for this Friday- but it may be beginning of next week before completed. Thanks! Cori


I wrote:

> I will update the support team. We will follow up the begging of next week to confirm the status and plan.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:58

@rranjan @ranand @amishra @Robleski 

I just want to confirm that once operations is able to update their accounts, then we will be good to release this? Or is there anything else that needs to be completed or looked into?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 15, 2020, 09:03

@kdjstudios @Robleski 

Thank you for the update.

* As per my understanding, we did not do any development yet we are just following the duplicate code issue fixing for all the sites.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 20, 2020, 15:52

Cori wrote:

> I am 90% complete with this. I handled it myself because I needed to research what the old codes were being used for & in some cases created new codes or decide to keep an older one. I expect to be done by Wed. Thanks Cori


I wrote:

> Please let us know when you have completed these updates. We will then have the support team confirm the updated universal billing codes are ready to be released and that there are no longer any accounts using codes that will be deactivated.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 21, 2020, 11:11

Cori wrote:

> * "Message Storage" 34100 isn't on the Master list of codes, but should be. So, didn't change these codes & added the code to the Master code list.
> * There were some accounts listed on here with the Code "Report Fee" (33062)- which is on the Master list.
> * There were some accounts listed on here with the Code “Setup/Programming (33064)- which is on the Master list.
> * There were some accounts listed on here with the Code “Hibernation Fee” (148)- which is on the Master list.
> * “Call Recording Storage” 33040 added to Master list.
> * “VCC In Call Agent Count” 55082- I added back to the Master list. There are some accounts using this and they are profitable.
> * “SMS Delivery” 33084 was added to Master List.
>  
> 
> A couple of general comments/questions:
> 
> * I’ve attached a new Master list.
> * I added the code # as part of the code name. Otherwise, there is no way to tell (by name only) what code was assigned to an account. If charges were generated for a specific code on an account, then it will show up on the Service Code Breakdown Summary report- however, if no  charges are generated, there’s no way to tell what the code # is to the code assigned. Please make sure the number is part of the name for all codes.
> * There are a few VCC Activity codes that I added back to the Master list. These are in use with some existing accounts at a few sites and they are profitable as they are, so I don’t want to change them. I would like to have these codes available only to the sites that are currently using them which I have listed in Column C of the Master code list. Ideally, they wouldn’t be able to be assigned to any new accounts even in these sites and only used for the accounts that already have them in use.
>  
> 
> All codes are reassigned. Please let me know if you find any stragglers. I came across one I didn’t see on the list that was sent. Thanks!
> 
> [All+Internal+VCC+Sites+-+Billing+Codes_CAB+July+2020.xlsx](/uploads/b998c97bb4ca207cd5c97ad8fc52aaa1/All+Internal+VCC+Sites+-+Billing+Codes_CAB+July+2020.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 28, 2020, 09:55

This was discussed today, Gary will want to have a meeting with Cori and Rich to make sure we handle this correctly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 11, 2020, 10:38

After reviewing these with the team, we are still wanting to keep a one posting program for all VCC sites. 

I will work with Jordan on getting a fresh dump of all codes that are no longer on the list, but are still active and in use. We will want to give Cori about 1-2 weeks to review the codes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 11, 2020, 16:39

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 12, 2020, 09:47

assigned to @aznbyte and unassigned @amishra

**Comment by kdjstudios** [comment]
In GitLab by @root on Aug 12, 2020, 10:38

Personal Notes for export > Site name, Account Name, and what Code is still in use that is not on the attached list. Do not include inactive items.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 1, 2020, 12:45

Cori wrote:

> Following up... Jayme & I spoke about this and also Gary & I last week. I just need the data dump showing all accounts that may still have a code we are not keeping on the Master List so that they can be removed. Once that's the done, the final step is just being sure that all of the old codes are deactivated in all sites and all of the new master codes are available in all sites. This is not new development, it's final clean up of a project that's 95% complete. Cori

@aznbyte Please add this to your queue to get that list of accounts and their codes for Cori.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 4, 2020, 12:58

@aznbyte 

I spent about 2 hours looking into the Studio 3T and with my limited knowledge of it, It seemed I would not be able to create a queue to do what we need to do for this task. I think this is due to the amount of logic and looping that needs to be done, since it is really a combination of multiple queries based upon results. 

So I switched over to looking at the current scripts that are being used by the cronjobs that we have running as I figured using something like that might work better for this, but we would just need to execute it once rather then setup as a cronjob. After looking at a few of the existing scripts, I felt comfortable enough to try luck at creating one to perform the task needed here. I believe I was able to create it, but I can not seem to get it to run. I also tried to manually run some of the other jobs and they wouldn't work either. So I think I might be missing a compile step or something somewhere, as the error I am receiving is `Your Ruby version is 1.9.3, but your Gemfile specified 2.5.0`.  


If you could look at what I created and see if anything jumps out as to why it isn't running. I have the files on `dev.sabilling.com`.

* `/home/sabilling/jobs/run_fee_finder.sh` - This file I believe is what should invokes/runs/compiles the ruby file. I pulled the basic code from the `run_ar_snapshot.sh` file. 
* `/home/sabilling/jobs/find_accounts_with_billing_codes.rb` - This file is the actual ruby code that should generate a list of the all results and then send an email with the results accordingly. This I used the `check_account_and_invoice_balance.rb` as a reference on how to structure the ruby code.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 4, 2020, 13:01

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @aznbyte on Sep 4, 2020, 14:31

@kdjstudios what we need to do is get RVM set up on your machine you can find it here: https://rvm.io/ This will let you set a ruby version for any code that you need to run. This will be useful in the future.

However I see that these are on the server, then we need to do a work around and I will take a look.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 9, 2020, 16:25

@aznbyte 

Thanks for the input on how to setup the RVM and use it. ALong with thanks for the aid on helping me use studio to find out why the code was not working as expected. 

I was able to get the list of all codes sent via email, which i can then copy over into excel and send to Cori. At some point I will want to get those setup to auto output into an csv file and just attach to the email. 

I wrote:

> Hi Cori, 
> 
> I have attached both the list of the All Sites Codes and the list of the accounts which still contain codes that are not on that list. Please review and let us know if you have any questions or concerns. 
> 
> [Billings_Codes.xlsx](/uploads/767325b2f58a52448a3debaf19e70267/Billings_Codes.xlsx)
> 
> [All+Internal+VCC+Sites+-+Billing+Codes_CAB__1___1_.xlsx](/uploads/a3579e58b64af5011e94fd325e3cfb83/All+Internal+VCC+Sites+-+Billing+Codes_CAB__1___1_.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 24, 2020, 08:58

@aznbyte @Robleski 

Cori wrote:

> Hi All- The data provided was not accurate/updated to reflect all Master codes and/or updates that had already been made - but I worked with it and have a final Master code list (attached).
> 
>  
> 
> Activities:               There are 2 to be added & 5 to be eliminated. They are noted as such on the attached.
> 
> Recurring/Other: There are 7 to be added.
> 
>  
> 
> There should be NO accounts which contain any codes not indicated (as being kept) on the attached Master List. I suggest deactivating the codes I indicated as no longer needed/in use sooner rather than later so they aren’t accidentally assigned.
> 
>  
> 
> To summarize next steps/finalization of this project:
> 
> All Codes (we are keeping) on the Master List should be available to all sites.
> All Codes existing in Sites NOT on the Master List should be removed from all sites.
> Ensure all Codes INCLUDE the CODE NUMBER in the NAME of the code as well.
> Ex:  “Message Storage 34100” and NOT “Message Storage”
>  
> 
> Please let me know if any questions. Thanks, Cori

[2020+SAB+MASTER+CODE+LIST+-+Updated+09+23+2020+CAB.xlsx](/uploads/240c24e0798b392fe279d553d7fad880/2020+SAB+MASTER+CODE+LIST+-+Updated+09+23+2020+CAB.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 24, 2020, 13:35

I was able to use the updated list to generate a new list of accounts that still have other billing codes on them. The list is attached. [SAB_VCC_Codes_Active.xlsx](/uploads/1ebf664d25badbc980db46155b2ef2ae/SAB_VCC_Codes_Active.xlsx)

I have also reached out to started to schedule a meeting to get Rich and Gary's approval on the updated list of billing codes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2020, 14:29

unassigned @aznbyte

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2020, 14:29

assigned to @kdjstudios

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2020, 14:50

I sent an update out to Cori on this. and hopefully we should have no more accounts using non-approved codes. Once that is completed @aznbyte and I should be able to update the posting scripts and the sites to use the approved codes. At that point we will want to review how Gary wants Ops/Cori to be notified when a site add fees not on the approved list.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 30, 2020, 14:49

After a few emails with Cori I was able to confirm that all sites on production are only using the approved codes.

[_2020+SAB+MASTER+CODE+LIST+-+Updated+09+29+20+CAB.xlsx](/uploads/0fef7b7ce8b57121bd8a1ed88fafc543/_2020+SAB+MASTER+CODE+LIST+-+Updated+09+29+20+CAB.xlsx)

I wrote:
> We were able to confirm at this point that all VCC Sites do not have any active accounts with active fees that are not on the approved list. 
> 
> The next steps are listed below.
> 1. [Completed On Branch] SA Billing Team to update the VCC Posting script to use the updated codes
> 2. [Needs to figure out Database entries for sites] SA Billing Team to update the HR report to use the updated codes
> 3. SA Billing Team and Cori to test and approve the updated posting script and HR report.
> 4. SA Billing Team to release the updates to all VCC sites on production. 
> 5. [In Progress] Update find script to actually deactivate those fees at the site level.

@aznbyte @Robleski 

I suggest we setup a meeting to review making the two updates above and how best to develop, test, and release them.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 8, 2020, 10:23

Was able to review this with @aznbyte yesterday. We were able to update the find fees script to not search for sites that are not VCC. I posted these updates to the github branch in the jobs repo.

I was able to creat a script that will deactivate all the fees at the account level and then also at the site level.

We will still need to create a script that will create all the fees that are missing at the site level for all the VCC Sites. As part of this script we should try to also have it update each site's VCC Posting script to V2.0.4 also

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 12, 2020, 13:16

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 15, 2020, 13:42

@Robleski @aznbyte 

I was able to test multiple time on the deactivate script and all is working as expected. I am hoping to have time later today to start working on the a script that will create the missing codes and also to update each site to V2.0.4 which should save us time from having to do it manually on 20ish sites.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2020, 08:48

I wrote:

> Just wanted to provide an update as Jordan has been moved to working on Open Answer bugs so this has not received as much attention as originally planned. However, I was able to spend time and was able to develop both 1 and 2 of the items below and they are currently being tested. One item I forgot to post on the list was to update all VCC sites to deactivate any remaining codes and to create any missing codes. This I am currently working on developing and hope to have it ready sometime next week. 
>  
> If all goes accordingly, we should be ready to do a full release of all these updates on a dev server by the start of the week of the 26th this month. I suspect we will need Cori and possibly some of operations perform some finalization tests/checks and then provide approval to release to actual production. 
>  
> One concern I have come across is that we have no way to prevent users who have access to the site edit page (Plus Ones/GMs) from reactivating any of the fees we deactivate or creating ones that are not on the approved list. As a solution, I thought maybe we could setup a daily/weekly check to find any fees that are not on the list and then send a notification email out if any were found. But would like to get your thoughts how we would like to handle this. 

Gary wrote:

> Lets discuss this at our meeting next week

Cori wrote:

> Hi Kyle- The daily check would be great. Thanks for the update. Cori

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 22, 2020, 13:38

@Robleski @aznbyte 

I have been able to get the git "sabilling_jobs" repo to sync on the dev server, however, I think it is only linked to my login. I didn't want to spend to much time researching how to get it synced under the main login since I don't have access to that yet. But overall I was able to start testing and syncing what will need to be done to get this synced with git before we try to do it on production. This should save us some time when we need to roll this out to production.

I was then able to setup the daily check scrip to run at roughly noon each day. I tested it manually today and it worked, but we should see by the time I come back form the long weekend if it works as expected automatically. 

I retested the deactivate codes and it appears to be working as expected too. We will want to run this on production when we release everything.

After researching on how to create any missing activities at a site and coming up with no easy viable solution. I think it would be best if we are to manually enter each activity on the site edit page that is missing. I was able to develop a script that will show all of the fees that each site has active after we run the deactivator script above. This should allow us to reference then which fees are missing and enter them.  

I also found that we may not have all of the exclude sites in the scripts. So we will want to re-reference production to determine a list of sites that are not VCC and will need to be excluded. 

As of right now, I will be following up with Cori to determine which site she would like to have setup on a dev server for her/ops to test with next week.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 22, 2020, 13:46

I wrote to Cori:

> We are testing a daily notice alert currently and will let you know once it is ready for you review. 
> 
> In regards to the overall release, we believe we are at a point where we are ready to allow you/ops to test and review to make sure all the updated codes are there and all the non approved codes are deactivated. To do so we would recommend to perform the following:
> 1. Check the site edit page and ensure only the billing codes (activities and additional fees) that are on the list show as active on  this page.
> 2. Run a full billing cycle to ensure all of the activities (calculated 5500 codes) calculate correctly  and generate invoices as expected. 
> 3. Once the billing cycle is ran, review the HR report and ensure the correct formula and results display and export as expected.
> 
> If you could provide us with which site you would like to perform the above tests and checks on, we will get it updated on a development server for your review. Thanks and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 27, 2020, 09:01

Cori wrote:

> Hi Kyle- Orlando please. Thanks Cori

I am working on catching up on the last two days while I was out on PTO, but once I am able to I will set this up for testing on a dev server.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 28, 2020, 12:27

@aznbyte @Robleski 

I was able to get caught up this morning and started to do a database back up from production and then restored it onto the dev server. This way we should have the dev server running in almost parallel to the production server.

At that point I ran the "fee deactivator" script to remove all fees that should no longer be active. This does it for all sites. This will generate an email with a list of all fees that were deactivated. 

After that I started to use Robo 3t to try and find a way to export entries in certain collections, then be able to modify those entries and then reimport the new ones. However Robo 3t from what I can tell does not allow us to do this. So this means we have to export every item individually for every site. Which is not a feasible option to do manually; seeing how we have 23 sites, and each site has about 20 activities and about 50 additional fees that needs to be setup. 

In addition to each site we then would need to setup the new version of the switch and each of the about 20 activities for the switch.  Once that is does we then need to point each site to the new switch. 

So @aznbyte with your knowledge of Mongo how can we accomplish this via a script/query to update all the sites and their activities and additional fees. Along with updating the switch to use the new activities too. I know the Aavaz team did not do this manually, but I can not find any script on any of the servers that do this, so I am hoping they shared that info with you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2020, 09:49

@aznbyte @Robleski 

After pairing with Jordan last week I was able to find out how to create new entries in the database. From this I was able to start designing the script to do the following:
1. Update each site's activities. This includes updating existing and entering new ones.
2. Update each site's additional fees. This includes updating existing and entering new ones. 
3. Update each site's switch tracker. This includes closing the last entry and adding a new entry to the new V2.0.4 switch.
4. Create new Switch version. This includes creating the new V2.0.4 switch and also creating new activities.

I believe I have created a script that will accomplish all of the above. I am now bringing testing on it to see if there are any adjustments or updates that will need to be done. If no issues, We should have this ready for Cori's testing this week.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2020, 15:13

I reached out to Cori to being testing on the updates.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 16, 2020, 13:03

changed title from **All {-Internal-} Sites - Billing Codes Updates** to **All {+VCC+} Sites - Billing Codes Updates**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 17, 2020, 15:04

1. What are our updated expectations on this now that Jordan is gone?
 -  I am a bit nervous about this, as I have only been learning Ruby and Mongo for 2 months. If something goes wrong, I am not very confident without help I would be able to resolve or fix it without impacts to the system. 
> Greg is comfortable with handling any backups and restores if issues occur. 
2. What is wrong with testing?
> Cori was using the wrong server. We should be good to go on testing again. Cori and Leah are performing the tests and expect to have feedback to us by next week.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2020, 14:50

Just got off the phone with Leah, she found we had a typo on the 55052 billing code, where it was showing the older "header label", I was able to update that on the dev server and re-run the script so Leah is able to continue testing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 10, 2020, 13:09

I wrote:

> I have been working on creating the documentation to send out to operations for the updates to the Billing codes and have it attached here. If you could please review it and provide your feedback or updates to the document would be greatly appreciated. 
> 
> Please note, that the only descriptions we have of the Activities are their calculations, so if someone could please provide the descriptions for those would be greatly appreciated. Also there were a few Additional Fees that were not part of the list when we came up with descriptions too, so if someone could please provide the descriptions for those would be greatly apricated too. 


[VCC_Universal_Billing_Codes.docx](/uploads/d85e623a7914e2bbf890724229c4fa35/VCC_Universal_Billing_Codes.docx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 11, 2020, 12:13

Fixed the issue with the 55084 billing code in the job and the posting script. Also fixed an issue with the release script which was creating duplicate switch versions on each site. Posting the updates to a new branch in github
